text/0008-styleguide.md

```diff
@@ -0,0 +1,91 @@
+flexible. Doing so calls on the distribution authors to come up with their own look and feel.
+
+OpenMRS could port [the old OpenMRS styleguide](https://demo.openmrs.org/openmrs/uicommons/styleGuide.page) into the new SPA and microfrontends.
+This would potentially provide an easier migration path for people using the old styleguide to switch their server rendered code to
+client rendered within the SPA. However, the old styleguide is not very well known and not used much or at all by many of the large distributions.
+Starting from scratch gives us a clean slate to make intentional and good decisions, for both the UX design and software development.
+
+The OpenMRS styleguide could be a custom build of an existing design library, such as Material, SemanticUI, and others.
+Using such a existing design library is discussed in [the Design Libraries RFC Proposal](/text/0010-design-libraries.md).
+
+The OpenMRS styleguide could not expose any global in-browser css custom properties (css variables). Instead, it could require all users of
+the styleguide to rely solely on css classes, or require all frontend code to use a css compiler such as SASS or LESS. Doing so doesn't make
+a ton of sense for OpenMRS, since all [OpenMRS supported browsers](/text/0003-browser-support.md) support css custom properties. See
+the [MDN compatibility table](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties#Browser_compatibility)
+for more information.
+
+The OpenMRS styleguide could not expose any global css classes, but rely entirely on javascript components. Doing so would give the maintainers
+of the styleguide more control over the user experience. However, it also complicates the styleguide code substantially, and raises the barrier
+to entry for using the styleguide. CSS classes are ubiquitous to all frontend code, whereas javascript components often take some time to
+set up and get going in everyone's framework and build process.
+
+The OpenMRS styleguide could not expose javascript components, but only css. This option would ensure the highest level of compatibility
+with javascript frameworks and codebases, since javascript components are much harder than css to use in all situations. See
+[the styleguide components proposal](/textt/0007-styleguide.components.md) for more discussion about this.
+
+## Common practices (not enforced)
+- Use existing styleguide css and components whenever possible during feature development.
+- Do not add too many things to the styleguide that become dead over time. Prefer a small set of things that are actively maintained
+  and heavily used. For example, a Button component is small enough that it can be easily maintained and heavily used,
+  whereas a searchable, sortable, paginated, filterable Table component is likely too hard to make fully customizable in a way
```

Comment:
I dunno, [DataTables](https://datatables.net/) are pretty fantastic...

Reply:
Yeah if we ever get to the point where we need documentation as shown below, then we've built something pretty crazy 😄 

*The quantity of documentation for a Data Table*

![datatables](https://user-images.githubusercontent.com/5524384/59216055-0bb10e00-8b78-11e9-8ff1-a2aabbf99a51.gif)